app/index.js

```diff
@@ -56,18 +61,57 @@ app.on('ready', () => {
     })
     // Set up crash reports.
     // Set up the needed stuff as the app launches.
-    ensuresIPFSInitialised()
-      .then(startIPFSDaemon)
-      .then((process) => {
-        console.log('IPFS Daemon started')
-        global.IPFS_PROCESS = process
-        loadingWindow.webContents.send('set-progress', {
-          text: 'Initializing IPFS client...',
-          percentage: 20
-        })
-        return Promise.resolve()
-      })
 
+    getAPIVersion()
+      .then(apiVersion => {
+        let customPorts = false
```

Comment:
Why don't we move this in another function with a promise returning true/false based on the answer of the user?